### PR TITLE
Flow reuse updates

### PR DESCRIPTION
**Pull request by karengermond**
#### PR Description

Create and add snippets where possible using the docs/shared shortcode as in https://github.com/grafana/agent/pull/2341.

#### Which issue(s) this PR fixes

https://github.com/grafana/agent/issues/2364

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
Hmm, some of the aliases seem wrong (like the cookbook ones have weird links, and some of the Flow ones are losing the Flow prefix), I'll need to go through them carefully on Monday to check

**Comment by karengermond** [comment]
> 

Sorry, I could create a new branch with the files I actually updated if that's easier.  Let me know.

File: docs/sources/shared/flow/reference/components/http-client-config-block.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+`http_client_config` block.
+
+The following sub-blocks are supported for `http_client_config`:
+
+Name | Description | Required
+---- | ----------- | --------
+[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against Kubernetes. | no
+[`authorization`](#authorization-block) | Configure generic authorization against Kubernetes. | no
+[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against Kubernetes. | no
+[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to Kubernetes. | no
```
Comment (rfratto) [inline docs/sources/shared/flow/reference/components/http-client-config-block.md:29]: nit: I think we stopped documenting subblocks in the content of a block now and we use the blocks header instead to describe the hierarchy. Should we remove this section?

**Review by rfratto — APPROVED** [review]
LGTM!

File: docs/sources/shared/flow/reference/components/http-client-config-block.md (hunk 1 above)
Comment (karengermond) [reply docs/sources/shared/flow/reference/components/http-client-config-block.md:29]: I'll update that in a following PR.

**Pull request merged**